visualization/app/codeCharta/ui/codeMap/codeMap.preRender.service.ts

```diff
@@ -54,9 +54,10 @@ export class CodeMapPreRenderService implements SettingsServiceSubscriber, FileS
 
 	public onSettingsChanged(settings: Settings, update: RecursivePartial<Settings>, event: angular.IAngularEvent) {
 		this.lastRender.settings = settings
-		if (this.lastRender.fileStates && update.fileSettings && update.fileSettings.blacklist) {
+		if (this.lastRender.fileStates && update.fileSettings && (update.fileSettings.blacklist || update.fileSettings.markedPackages)) {
 			this.lastRender.renderFile = this.getSelectedFilesAsUnifiedMap(this.lastRender.fileStates)
-			this.lastRender.renderFile.settings.fileSettings = settings.fileSettings
```

Comment:
Why is setting the whole fileSettings object a bad idea?

Reply:
I gonna check again, but I remember, that the attributeTypes got deleted when overwriting the with the `settings.fileSettings`, because they didn't exist in there for some reason